go.mod

```diff
@@ -3,11 +3,13 @@ module github.com/ceph/ceph-csi-operator
 go 1.21
 
 require (
+	github.com/go-logr/logr v1.4.1
 	github.com/onsi/ginkgo/v2 v2.14.0
 	github.com/onsi/gomega v1.30.0
 	k8s.io/api v0.29.2
 	k8s.io/apimachinery v0.29.2
 	k8s.io/client-go v0.29.2
+	k8s.io/utils v0.0.0-20230726121419-3b25d923346b
 	sigs.k8s.io/controller-runtime v0.17.3
```

Comment:
we should be working with controller-runtime v0.18 and with other latest modules. I'm working on rook to upgrade to v0.18 it requires some changes to upgrade from 0v.17 so it is better to start with the latest modules.

@Madhu-1 thoughts

Reply:
also with k8s v0.30.2 and go 1.22

Reply:
Sure, will take care of dependency updates before release.

Reply:
@subhamkrai I used the stable release that was available when I generated the project. If updates are needed we will bump after we have the code up and running.

Reply:
@subhamkrai as this is not the project scaffolding PR (that one already merged) and your request does not tie directly into the code changes in this PR, I will advise opening a git hub issue to track the request for the upgrade that will be addressed in a future PR.

With that in mind, I am resolving this thread.